library/Fission/AWS/Route53.hs

```diff
@@ -3,30 +3,66 @@ module Fission.AWS.Route53
   , registerDomain
   ) where
 
-import Network.AWS
-import Network.AWS.Auth as AWS
-import Network.AWS.Prelude
-import Network.AWS.Route53
-import Servant
+import           Network.AWS
+import           Network.AWS.Auth as AWS
+import           Network.AWS.Prelude hiding (hash)
+import           Network.AWS.Route53
+import           Servant
 
 import           Fission.Prelude
+import           Fission.Internal.UTF8
 import qualified Fission.Config    as Config
 import           Fission.AWS.Types as AWS
 import           Fission.AWS
 
-registerDomain
-  :: ( MonadRIO           cfg m
-     , MonadUnliftIO          m
-     , HasLogFunc         cfg
-     , Has AWS.AccessKey  cfg
-     , Has AWS.SecretKey  cfg
-     , Has AWS.ZoneID     cfg
-     )
+import qualified Fission.IPFS.Types as IPFS
+import           Fission.IPFS.Gateway.Types
+import           Fission.IPFS.CID.Types
+
+registerDomain ::
+  ( MonadRIO           cfg m
+  , MonadUnliftIO          m
+  , HasLogFunc         cfg
+  , Has IPFS.Gateway   cfg
+  , Has AWS.AccessKey  cfg
+  , Has AWS.SecretKey  cfg
+  , Has AWS.ZoneID     cfg
+  , Has AWS.DomainName cfg
+  )
+  => Text
+  -> CID
+  -> m (Either ServerError AWS.DomainName)
+registerDomain username (CID hash) = do
+  gateway :: IPFS.Gateway   <- Config.get
+  domain  :: AWS.DomainName <- Config.get
+
+  let
+    baseUrl    = username <> "."<> AWS.getDomainName domain
+    dnslinkUrl = "_dnslink." <> baseUrl
+    dnslink    = "dnslink=/ipfs/" <> hash
+
+  logDebug <| displayShow ("~~" <> (wrapIn dnslink "\"") <> "~~")
```

Comment:
Totally fine as-is, but cool trick to make a function infix (reads like pseudo-English)

```haskell
logDebug <| displayShow ("~~" <> dnslink `wrapIn` "\"" <> "~~")
```